Jenkinsfile

```diff
@@ -26,18 +26,12 @@ def sendEmailNotifications() {
     )
 }
 
-def getPublishingRepository() {
-    def gitBranch = env.BRANCH_NAME // https://issues.jenkins-ci.org/browse/JENKINS-30252
-    if(gitBranch.startsWith("release")) {
-        return 'Release'
-    }
-    if(gitBranch == "master") {
-        return 'Snapshot'
-    }
-    return null
+def isPublishNeeded() {
+    def gitBranch = env.BRANCH_NAME
+    return gitBranch.startsWith("release") || gitBranch == "master"
 }
 
-def BuildVersionX(publishingRepo, version) {
+def BuildVersionX(isPublishNeeded, version) {
```

Comment:
I would avoid shadowing the function with a boolean, I got pretty confused thinking why you would want to pass around a function here.

Reply:
Ok, will adjust.